fgpyo/sam/__init__.py

```diff
@@ -876,6 +876,78 @@ def set_mate_info(
     r2.is_proper_pair = proper_pair
 
 
+def set_mate_info_on_secondary(
+    primary: AlignedSegment,
+    secondary: AlignedSegment,
+    is_proper_pair: Callable[[AlignedSegment, AlignedSegment], bool] = is_proper_pair,
+) -> None:
+    """Set mate info on a secondary alignment to the next read ordinal's primary alignment.
+
+    Args:
+        primary: The primary alignment of the secondary's mate.
+        secondary: The secondary alignment to set mate information upon.
+        is_proper_pair: A function that takes the two alignments and determines proper pair status.
+
+    Raises:
+        ValueError: If primary and secondary are of the same read ordinal.
+        ValueError: If the primary is marked as either secondary or supplementary.
+        ValueError: If the secondary is not marked as secondary.
+    """
+    if primary.is_read1 == secondary.is_read1 or primary.is_secondary or primary.is_supplementary:
+        raise ValueError("Secondary mate info must be set from a primary of the next ordinal!")
+    if not secondary.is_secondary:
+        raise ValueError("Cannot set mate info on an alignment not marked as secondary!")
+    if primary.query_name != secondary.query_name:
+        raise ValueError("Cannot set mate info on alignments with different query names!")
+
+    secondary.next_reference_id = primary.reference_id
+    secondary.next_reference_name = primary.reference_name
+    secondary.next_reference_start = primary.reference_start
+    secondary.mate_is_forward = primary.is_forward
+    secondary.mate_is_mapped = primary.is_mapped
+    secondary.set_tag("MC", primary.cigarstring)
+    secondary.set_tag("MQ", primary.mapping_quality)
+    secondary.set_tag("ms", sum_of_base_qualities(primary))
```

Comment:
Could all of this be extracted into a `_transfer_mate_fields()` that is used by this function, `set_mate_info()` and also `set_mate_info_on_supplementary()`; i.e. put all the one-way transfer into a function and re-use that.